### DISCUSSION TITLE
Feedback and Discussion of MMEditing 1.x

### DISCUSSION THREAD

**Discussion by vansin** [open-mmlab/mmediting#1108] [General]
Hi all, we have released MMEditing 1.x 🎉🎉🎉

## MMEditing 1.x has merged MMGeneration!

Therefore, MMEditing 1.x now has 
- Supported 42+ algorithms, 169+ configs, and 168+ checkpoints.
- Supported 26+ loss functions, 20+ metrics.
- Supported tensorboard, wandb.
- Supported unconditional GANs, conditional GANs, image2image translation, and internal learning.

### Release

You can find the latest release of MMEditing 1.x here: https://github.com/open-mmlab/mmediting/releases/tag/v1.0.0rc1

### Stable version

You can check the code of the stable version of MMEditing 1.x here: https://github.com/open-mmlab/mmediting/tree/1.x. We usually update this branch once a month. 

### Developing version

You can check the developing branch of MMEditing 1.x here: https://github.com/open-mmlab/mmediting/tree/dev-1.x. We develop MMEditing 1.x based on dev-1.x branch. 

### Documentation

You can find the documentation of MMEditing 1.x here: https://mmediting.readthedocs.io/en/1.x/

All [suggestions](https://github.com/open-mmlab/mmediting/issues), [contributions](https://github.com/open-mmlab/mmediting/pulls), and [discussions](https://github.com/open-mmlab/mmediting/discussions/1108) are welcome! 💪
We will try our best to response every issue, pull request and discussion within 24 hours 🥰🥰🥰

**Comment by zengyh1900** [comment]
Hi, here is a comment from Bo. 

> 'The documentation is not friendly enough for me to learn MMEditing 1.x. For example, I have to check the source code if I want to know where is the `collect` in the original `pipeline`'

Hi, we have provided the migration documentation here https://mmediting.readthedocs.io/en/1.x/migration/6_data.html. `Collect` and `ToTensor` has been combined as `PackEditInputs`.

**Comment by zengyh1900** [comment]
A comment from Bo,

1. Why does mmcv not include `PackEditInputs`? Does it mean that all the repos need to implement their own `PackEditInputs`? It seems not better than `Collect`, which can be used by all the repos. For example, if someone needs to import both mmedit and mmseg, which PackInput should he use? PackSegInputs or PackEditInputs? Or he should implement a new one?

**Comment by zengyh1900** [comment]
A comment from Ruoning,

> 1. 嗯嗯，感觉差异还是挺明显的，学习门槛也确实高了一些，印象比较深的几个地方就是加了mmengine之后出现了很多新的特性，尤其是engine里面的registry.switch_scope_and_registry当时就感觉挺抽象的[笑哭]，
> 2. 然后就是dataloader那块的edit_data_sample里面放了好多fake_img、pred_img之类的，我当时以为collect的东西需要在这里面添加，也是试了几次才清楚用法，这块文档里面确实也没有涉及。
> 3. 感觉可以在迁移的部分或者构建模型的部分，添加一些容易出现问题的点或者注意事项之类的。涉及到mmengine或者mmcv里面的一些高级的用法、抽象的设计也可以加个跳转链接到他们的文档下

**Comment by cgwu1999** [comment]
I met a problem, I trained a model with mmediting and saved the params, then I tested the params in basicsr, the ssim metric value is 0.3 higher than in mmediting. I found they get different outputs with the same input and params. i think it is a problem with the mmdistdataparall. Because I found the result is different using model.forward(this is a distdataparall) and model.module.forward.

**Reply by zengyh1900** [reply]
Hi @PiKaChu-wcg , thank you for pointing out this issue! I will check it today and update it to you! If you have any additional information, please kindly tell us!

**Comment by likakakaka** [comment]
TensorboardLoggerHook has been changed in the new version. How can I see the training process by tensorboard in the new version?

**Comment by Z-Fran** [comment]
Please use `TensorboardVisBackend` following https://mmengine.readthedocs.io/en/latest/common_usage/visualize_training_log.html and https://mmagic.readthedocs.io/en/latest/user_guides/visualization.html . @likakakaka 

> TensorboardLoggerHook has been changed in the new version. How can I see the training process by tensorboard in the new version?
